src/data_provider/src/extended_sources/browser_extensions/src/chrome.cpp

```diff
@@ -0,0 +1,685 @@
+        result.reserve(length * 2); // two letters per byte (high and low nibble)
+
+        for (size_t i = 0; i < length; ++i)
+        {
+            uint8_t byte = hash[i];
+            // high nibble
+            result.push_back('a' + ((byte >> 4) & 0x0F));
+            // low nibble
+            result.push_back('a' + (byte & 0x0F));
+        }
+
+        return result;
+    }
+
+    std::string ChromeExtensionsProvider::base64Decode(const std::string& input)
+    {
+        static const std::string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+        std::string decoded;
+        std::vector<int> T(256, -1);
+
+        // Build lookup table
+        for (int i = 0; i < 64; i++) T[chars[i]] = i;
+
+        int val = 0, valb = -8;
+
+        for (unsigned char c : input)
+        {
+            if (T[c] == -1) break;
+
+            val = (val << 6) + T[c];
+            valb += 6;
+
+            if (valb >= 0)
+            {
+                decoded.push_back(char((val >> valb) & 0xFF));
+                valb -= 8;
+            }
+        }
+
+        return decoded;
+    }
+
+    std::string ChromeExtensionsProvider::generateIdentifier(const std::string& key)
+    {
+        // Decode to string first
+        std::string decodedString = base64Decode(key);
+
+        // Convert to vector<uint8_t> to match original behavior exactly
+        std::vector<uint8_t> decodedVector(decodedString.begin(), decodedString.end());
+
+        uint8_t hash[SHA256_DIGEST_LENGTH];
+        SHA256(decodedVector.data(), decodedVector.size(), hash);
+
+        std::string letters_string = hashToLetterString(hash, SHA256_DIGEST_LENGTH);
+        return letters_string.substr(0, 32);
+    }
+
+    std::string ChromeExtensionsProvider::hashToHexString(const uint8_t* hash, size_t length)
+    {
+        std::ostringstream oss;
+        oss << std::hex << std::setfill('0');
+
+        for (size_t i = 0; i < length; ++i)
+        {
+            oss << std::setw(2) << static_cast<int>(hash[i]);
+        }
+
+        return oss.str();
+    }
+
+    std::string ChromeExtensionsProvider::sha256File(const std::string& filepath)
+    {
+        std::ifstream file(filepath, std::ios::binary);
+
+        if (!file)
+        {
+            throw std::runtime_error("Cannot open file: " + filepath);
+        }
+
+        SHA256_CTX sha256;
+        SHA256_Init(&sha256);
+
+        std::vector<char> buffer(8192);
+
+        while (file.read(buffer.data(), buffer.size()) || file.gcount() > 0)
+        {
+            SHA256_Update(&sha256, buffer.data(), file.gcount());
+        }
+
+        uint8_t hash[SHA256_DIGEST_LENGTH];
+        SHA256_Final(hash, &sha256);
+
+        // Reuse the same hex function if needed
+        return hashToHexString(hash, SHA256_DIGEST_LENGTH);
+    }
+
+    std::string ChromeExtensionsProvider::webkitToUnixTime(std::string webkit_timestamp)
+    {
+        int64_t timestamp = std::stoll(webkit_timestamp);
```

Comment:
Timestamp Conversion Without Validation

```suggestion
  try
  {
      if (webkit_timestamp.empty() || !std::all_of(webkit_timestamp.begin(), webkit_timestamp.end(), ::isdigit))
      {
          return "0";
      }

      int64_t timestamp = std::stoll(webkit_timestamp);

      // Validar rango (WebKit epoch válido)
      if (timestamp < 11644473600000000LL || timestamp > 253402300799000000LL)
      {
          return "0";
      }

      std::time_t unix_timestamp = (timestamp - 11644473600000000LL) / 1000000;
      return std::to_string(unix_timestamp);
  }
  catch (const std::exception& e)
  {
      return "0";
  }

```